### PR TITLE
Extracted Authentication and introduced more flexibility

### PR DESCRIPTION
**Pull request by parndt**
This extracts authentication and implements a basic authentication/authorisation adapter in the core.

Could everyone please review this to determine what is likely to cause problems so that we can get that sorted?

### CODE REVIEW

**Comment by bricesanchez** [comment]
i will test it for my new spree/refinery project

**Comment by bricesanchez** [comment]
At this time, i'm trying to tell refinerycms to use the User model of Spree, could you help me with that ?

BTW, it's pretty stable and easier to use refinerycms with spree... :smile:

File: core/lib/generators/refinery/form/templates/app/models/refinery/namespace/setting.rb.erb [hunk 1]
```diff
@@ -1,3 +1,5 @@
+require 'zilch/authorisation/users_manager'
```
Comment (parndt) [inline core/lib/generators/refinery/form/templates/app/models/refinery/namespace/setting.rb.erb]: This isn't required if this is required in the `Refinery::Core::BaseModel` already.

**Comment by bricesanchez** [comment]
This is my Gemfile : 
```ruby
source 'https://rubygems.org'


# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
gem 'rails', '4.2.0'
# Use sqlite3 as the database for Active Record

gem 'pg'

gem 'puma'

# Use SCSS for stylesheets
gem 'sass-rails', '~> 5.0'
# Use Uglifier as compressor for JavaScript assets
gem 'uglifier', '>= 1.3.0'
# Use CoffeeScript for .coffee assets and views
gem 'coffee-rails', '~> 4.1.0'
# See https://github.com/sstephenson/execjs#readme for more supported runtimes
# gem 'therubyracer', platforms: :ruby

# Use jquery as the JavaScript library
gem 'jquery-rails'
# Turbolinks makes following links in your web application faster. Read more: https://github.com/rails/turbolinks
gem 'turbolinks'
gem 'jquery-turbolinks'

# Slim lang
gem 'slim'

# Neat bourbon RWD Framework
gem 'neat'
gem 'bourbon'
gem 'bitters'

gem 'bootstrap-sass'

# Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
gem 'jbuilder', '~> 2.0'
# bundle exec rake doc:rails generates the API under doc/api.
gem 'sdoc', '~> 0.4.0', group: :doc

# Use ActiveModel has_secure_password
# gem 'bcrypt', '~> 3.1.7'

# Use Unicorn as the app server
# gem 'unicorn'

# Use Capistrano for deployment
# gem 'capistrano-rails', group: :development

group :development, :test do
  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
  gem 'byebug'

  # Access an IRB console on exception pages or by using <%= console %> in views
  gem 'web-console', '~> 2.0'

  # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
  gem 'spring'

  gem 'letter_opener'
  gem 'sextant'
  gem 'guard', '>= 2.2.2',       :require => false
  gem 'guard-livereload',        :require => false
  gem 'rack-livereload'
  gem 'rb-fsevent',              :require => false
  gem 'figaro'
end

group :production do
  # Heroku
  gem 'dragonfly-s3_data_store'
  gem 'rails_12factor'
  gem 'rails_on_heroku'
  gem 'newrelic_rpm'

  # Caching strategy
  gem 'actionpack-page_caching'
  gem 'rack-cache'
  gem 'dalli'
  gem 'kgio'
end


gem 'refinerycms',         github: 'parndt/refinerycms',         branch: 'auth-for-real-yo-tmp'
gem 'zilch-authorisation', github: 'parndt/zilch-authorisation', branch: "master"

gem 'quiet_assets', :group => :development

# Add support for refinerycms-acts-as-indexed
gem 'refinerycms-acts-as-indexed', ['~> 2.0', '>= 2.0.0']

# Add support for refinerycms-wymeditor
gem 'refinerycms-wymeditor', ['~> 1.0', '>= 1.0.6']

# gem 'refinerycms-blog',        github: 'refinery/refinerycms-blog',        branch: 'master'
gem 'refinerycms-inquiries',   github: 'refinery/refinerycms-inquiries',   branch: 'master'
gem 'refinerycms-page-images', github: 'refinery/refinerycms-page-images', branch: 'master'
gem 'refinerycms-settings',    github: 'refinery/refinerycms-settings',    branch: 'master'
gem 'refinerycms-search',      github: 'refinery/refinerycms-search',      branch: 'master'

gem 'refinerycms-retailers', github: 'bisscomm/refinerycms-retailers', branch: '3-0-stable'
gem 'refinerycms-jobs',      github: 'bisscomm/refinerycms-jobs',      branch: '3-0-stable'


gem 'spree',             github: 'spree/spree',             branch: '3-0-stable'
gem 'spree_i18n',        github: 'spree/spree_i18n',        branch: '3-0-stable'
gem 'spree_auth_devise', github: 'spree/spree_auth_devise', branch: '3-0-stable'

gem 'spree-refinerycms-authentication', github: 'bricesanchez/spree-refinery-authentication', branch: 'auth-for-real-yo-tmp'
```

This is my current work to make Spree and Refinery compatible : 
https://github.com/bricesanchez/spree-refinery-authentication/tree/auth-for-real-yo-tmp

**Comment by parndt** [comment]
Thanks @bricesanchez !

**Comment by bricesanchez** [comment]
At this time, my error is here : 
```ruby
ActionView::Template::Error (undefined method `plugins' for #<Spree::User:0x007fd302539a80>):
    1: <nav id="menu">
    2:   <%= render partial: "refinery/admin/menu_item",
    3:              collection: current_refinery_user.plugins.in_menu -%>
    4: </nav>
  .gems/gems/activemodel-4.2.0/lib/active_model/attribute_methods.rb:433:in `method_missing'
  .gems/bundler/gems/refinerycms-6ad393d84808/core/app/views/refinery/admin/_menu.html.erb:3:in `_7858e449e7d6c92ed155b66e9c2077db'
  .gems/gems/actionview-4.2.0/lib/action_view/template.rb:145:in `block in render'
  .gems/gems/activesupport-4.2.0/lib/active_support/notifications.rb:166:in `instrument'
  .gems/gems/actionview-4.2.0/lib/action_view/template.rb:333:in `instrument'
  .gems/gems/actionview-4.2.0/lib/action_view/template.rb:143:in `render'
  .gems/gems/deface-1.0.1/lib/deface/action_view_extensions.rb:41:in `render'
  .gems/gems/actionview-4.2.0/lib/action_view/renderer/partial_renderer.rb:339:in `render_partial'
  .gems/gems/actionview-4.2.0/lib/action_view/renderer/partial_renderer.rb:310:in `block in render'
  .gems/gems/actionview-4.2.0/lib/action_view/renderer/abstract_renderer.rb:39:in `block in instrument'
  .gems/gems/activesupport-4.2.0/lib/active_support/notifications.rb:164:in `block in instrument'
  .gems/gems/activesupport-4.2.0/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
  .gems/gems/activesupport-4.2.0/lib/active_support/notifications.rb:164:in `instrument'
  .gems/gems/actionview-4.2.0/lib/action_view/renderer/abstract_renderer.rb:39:in `instrument'
  .gems/gems/actionview-4.2.0/lib/action_view/renderer/partial_renderer.rb:309:in `render'
  .gems/gems/actionview-4.2.0/lib/action_view/renderer/renderer.rb:47:in `render_partial'
  .gems/gems/actionview-4.2.0/lib/action_view/helpers/rendering_helper.rb:35:in `render'
  .gems/bundler/gems/refinerycms-6ad393d84808/core/app/views/layouts/refinery/admin.html.erb:10:in `_28f0cef288359a4cf2396cebb0b0cff8'
  .gems/gems/actionview-4.2.0/lib/action_view/template.rb:145:in `block in render'
  .gems/gems/activesupport-4.2.0/lib/active_support/notifications.rb:166:in `instrument'
  .gems/gems/actionview-4.2.0/lib/action_view/template.rb:333:in `instrument'
  .gems/gems/actionview-4.2.0/lib/action_view/template.rb:143:in `render'
  .gems/gems/deface-1.0.1/lib/deface/action_view_extensions.rb:41:in `render'
  .gems/gems/actionview-4.2.0/lib/action_view/renderer/template_renderer.rb:66:in `render_with_layout'
  .gems/gems/actionview-4.2.0/lib/action_view/renderer/template_renderer.rb:52:in `render_template'
  .gems/gems/actionview-4.2.0/lib/action_view/renderer/template_renderer.rb:14:in `render'
  .gems/gems/actionview-4.2.0/lib/action_view/renderer/renderer.rb:42:in `render_template'
  .gems/gems/actionview-4.2.0/lib/action_view/renderer/renderer.rb:23:in `render'
  .gems/gems/actionview-4.2.0/lib/action_view/rendering.rb:100:in `_render_template'
  .gems/gems/actionpack-4.2.0/lib/action_controller/metal/streaming.rb:217:in `_render_template'
  .gems/gems/actionview-4.2.0/lib/action_view/rendering.rb:83:in `render_to_body'
  .gems/gems/actionpack-4.2.0/lib/action_controller/metal/rendering.rb:32:in `render_to_body'
  .gems/gems/actionpack-4.2.0/lib/action_controller/metal/renderers.rb:37:in `render_to_body'
  .gems/gems/actionpack-4.2.0/lib/abstract_controller/rendering.rb:25:in `render'
  .gems/gems/actionpack-4.2.0/lib/action_controller/metal/rendering.rb:16:in `render'
  .gems/gems/actionpack-4.2.0/lib/action_controller/metal/instrumentation.rb:41:in `block (2 levels) in render'
  .gems/gems/activesupport-4.2.0/lib/active_support/core_ext/benchmark.rb:12:in `block in ms'
  /Users/bci20/.rbenv/versions/2.2.0/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
  .gems/gems/activesupport-4.2.0/lib/active_support/core_ext/benchmark.rb:12:in `ms'
  .gems/gems/actionpack-4.2.0/lib/action_controller/metal/instrumentation.rb:41:in `block in render'
  .gems/gems/actionpack-4.2.0/lib/action_controller/metal/instrumentation.rb:84:in `cleanup_view_runtime'
  .gems/gems/activerecord-4.2.0/lib/active_record/railties/controller_runtime.rb:25:in `cleanup_view_runtime'
  .gems/gems/actionpack-4.2.0/lib/action_controller/metal/instrumentation.rb:40:in `render'
  .gems/gems/actionpack-4.2.0/lib/action_controller/metal/implicit_render.rb:10:in `default_render'
  .gems/gems/actionpack-4.2.0/lib/action_controller/metal/implicit_render.rb:5:in `send_action'
  .gems/gems/actionpack-4.2.0/lib/abstract_controller/base.rb:198:in `process_action'
  .gems/gems/actionpack-4.2.0/lib/action_controller/metal/rendering.rb:10:in `process_action'
  .gems/gems/actionpack-4.2.0/lib/abstract_controller/callbacks.rb:20:in `block in process_action'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:117:in `call'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:117:in `call'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:219:in `block in halting_and_conditional'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:169:in `call'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:169:in `block in halting'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:169:in `call'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:169:in `block in halting'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:169:in `call'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:169:in `block in halting'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:234:in `call'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:234:in `block in halting'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:234:in `call'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:234:in `block in halting'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:169:in `call'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:169:in `block in halting'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:169:in `call'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:169:in `block in halting'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:169:in `call'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:169:in `block in halting'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:169:in `call'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:169:in `block in halting'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:169:in `call'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:169:in `block in halting'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:169:in `call'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:169:in `block in halting'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:151:in `call'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:151:in `block in halting_and_conditional'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:151:in `call'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:151:in `block in halting_and_conditional'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:92:in `call'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:92:in `_run_callbacks'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:734:in `_run_process_action_callbacks'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:81:in `run_callbacks'
  .gems/gems/actionpack-4.2.0/lib/abstract_controller/callbacks.rb:19:in `process_action'
  .gems/gems/actionpack-4.2.0/lib/action_controller/metal/rescue.rb:29:in `process_action'
  .gems/gems/actionpack-4.2.0/lib/action_controller/metal/instrumentation.rb:31:in `block in process_action'
  .gems/gems/activesupport-4.2.0/lib/active_support/notifications.rb:164:in `block in instrument'
  .gems/gems/activesupport-4.2.0/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
  .gems/gems/activesupport-4.2.0/lib/active_support/notifications.rb:164:in `instrument'
  .gems/gems/actionpack-4.2.0/lib/action_controller/metal/instrumentation.rb:30:in `process_action'
  .gems/gems/actionpack-4.2.0/lib/action_controller/metal/params_wrapper.rb:250:in `process_action'
  .gems/gems/activerecord-4.2.0/lib/active_record/railties/controller_runtime.rb:18:in `process_action'
  .gems/gems/actionpack-4.2.0/lib/abstract_controller/base.rb:137:in `process'
  .gems/gems/actionview-4.2.0/lib/action_view/rendering.rb:30:in `process'
  .gems/gems/actionpack-4.2.0/lib/action_controller/metal.rb:195:in `dispatch'
  .gems/gems/actionpack-4.2.0/lib/action_controller/metal/rack_delegation.rb:13:in `dispatch'
  .gems/gems/actionpack-4.2.0/lib/action_controller/metal.rb:236:in `block in action'
  .gems/gems/actionpack-4.2.0/lib/action_dispatch/routing/route_set.rb:73:in `call'
  .gems/gems/actionpack-4.2.0/lib/action_dispatch/routing/route_set.rb:73:in `dispatch'
  .gems/gems/actionpack-4.2.0/lib/action_dispatch/routing/route_set.rb:42:in `serve'
  .gems/gems/actionpack-4.2.0/lib/action_dispatch/journey/router.rb:43:in `block in serve'
  .gems/gems/actionpack-4.2.0/lib/action_dispatch/journey/router.rb:30:in `each'
  .gems/gems/actionpack-4.2.0/lib/action_dispatch/journey/router.rb:30:in `serve'
  .gems/gems/actionpack-4.2.0/lib/action_dispatch/routing/route_set.rb:802:in `call'
  .gems/gems/railties-4.2.0/lib/rails/engine.rb:518:in `call'
  .gems/gems/railties-4.2.0/lib/rails/railtie.rb:194:in `public_send'
  .gems/gems/railties-4.2.0/lib/rails/railtie.rb:194:in `method_missing'
  .gems/gems/actionpack-4.2.0/lib/action_dispatch/routing/mapper.rb:51:in `serve'
  .gems/gems/actionpack-4.2.0/lib/action_dispatch/journey/router.rb:43:in `block in serve'
  .gems/gems/actionpack-4.2.0/lib/action_dispatch/journey/router.rb:30:in `each'
  .gems/gems/actionpack-4.2.0/lib/action_dispatch/journey/router.rb:30:in `serve'
  .gems/gems/actionpack-4.2.0/lib/action_dispatch/routing/route_set.rb:802:in `call'
  .gems/gems/dragonfly-1.0.7/lib/dragonfly/middleware.rb:14:in `call'
  .gems/gems/dragonfly-1.0.7/lib/dragonfly/middleware.rb:14:in `call'
  .gems/gems/dragonfly-1.0.7/lib/dragonfly/middleware.rb:14:in `call'
  .gems/gems/warden-1.2.3/lib/warden/manager.rb:35:in `block in call'
  .gems/gems/warden-1.2.3/lib/warden/manager.rb:34:in `catch'
  .gems/gems/warden-1.2.3/lib/warden/manager.rb:34:in `call'
  .gems/gems/rack-1.6.0/lib/rack/etag.rb:24:in `call'
  .gems/gems/rack-1.6.0/lib/rack/conditionalget.rb:25:in `call'
  .gems/gems/rack-1.6.0/lib/rack/head.rb:13:in `call'
  .gems/gems/actionpack-4.2.0/lib/action_dispatch/middleware/params_parser.rb:27:in `call'
  .gems/gems/actionpack-4.2.0/lib/action_dispatch/middleware/flash.rb:260:in `call'
  .gems/gems/rack-1.6.0/lib/rack/session/abstract/id.rb:225:in `context'
  .gems/gems/rack-1.6.0/lib/rack/session/abstract/id.rb:220:in `call'
  .gems/gems/actionpack-4.2.0/lib/action_dispatch/middleware/cookies.rb:560:in `call'
  .gems/gems/activerecord-4.2.0/lib/active_record/query_cache.rb:36:in `call'
  .gems/gems/activerecord-4.2.0/lib/active_record/connection_adapters/abstract/connection_pool.rb:647:in `call'
  .gems/gems/activerecord-4.2.0/lib/active_record/migration.rb:378:in `call'
  .gems/gems/actionpack-4.2.0/lib/action_dispatch/middleware/callbacks.rb:29:in `block in call'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:88:in `call'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:88:in `_run_callbacks'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:734:in `_run_call_callbacks'
  .gems/gems/activesupport-4.2.0/lib/active_support/callbacks.rb:81:in `run_callbacks'
  .gems/gems/actionpack-4.2.0/lib/action_dispatch/middleware/callbacks.rb:27:in `call'
  .gems/gems/actionpack-4.2.0/lib/action_dispatch/middleware/reloader.rb:73:in `call'
  .gems/gems/actionpack-4.2.0/lib/action_dispatch/middleware/remote_ip.rb:78:in `call'
  .gems/gems/actionpack-4.2.0/lib/action_dispatch/middleware/debug_exceptions.rb:17:in `call'
  .gems/gems/web-console-2.1.2/lib/web_console/middleware.rb:37:in `call'
  .gems/gems/actionpack-4.2.0/lib/action_dispatch/middleware/show_exceptions.rb:30:in `call'
  .gems/gems/railties-4.2.0/lib/rails/rack/logger.rb:38:in `call_app'
  .gems/gems/railties-4.2.0/lib/rails/rack/logger.rb:20:in `block in call'
  .gems/gems/activesupport-4.2.0/lib/active_support/tagged_logging.rb:68:in `block in tagged'
  .gems/gems/activesupport-4.2.0/lib/active_support/tagged_logging.rb:26:in `tagged'
  .gems/gems/activesupport-4.2.0/lib/active_support/tagged_logging.rb:68:in `tagged'
  .gems/gems/railties-4.2.0/lib/rails/rack/logger.rb:20:in `call'
  .gems/gems/quiet_assets-1.1.0/lib/quiet_assets.rb:27:in `call_with_quiet_assets'
  .gems/gems/actionpack-4.2.0/lib/action_dispatch/middleware/request_id.rb:21:in `call'
  .gems/gems/rack-1.6.0/lib/rack/methodoverride.rb:22:in `call'
  .gems/gems/rack-1.6.0/lib/rack/runtime.rb:18:in `call'
  .gems/gems/activesupport-4.2.0/lib/active_support/cache/strategy/local_cache_middleware.rb:28:in `call'
  .gems/gems/rack-1.6.0/lib/rack/lock.rb:17:in `call'
  .gems/gems/dragonfly-1.0.7/lib/dragonfly/cookie_monster.rb:9:in `call'
  .gems/gems/rack-livereload-0.3.15/lib/rack/livereload.rb:23:in `_call'
  .gems/gems/rack-livereload-0.3.15/lib/rack/livereload.rb:14:in `call'
  .gems/gems/actionpack-4.2.0/lib/action_dispatch/middleware/static.rb:113:in `call'
  .gems/gems/rack-1.6.0/lib/rack/sendfile.rb:113:in `call'
  .gems/gems/railties-4.2.0/lib/rails/engine.rb:518:in `call'
  .gems/gems/railties-4.2.0/lib/rails/application.rb:164:in `call'
  .gems/gems/rack-1.6.0/lib/rack/content_length.rb:15:in `call'
  .gems/gems/puma-2.11.1/lib/puma/server.rb:507:in `handle_request'
  .gems/gems/puma-2.11.1/lib/puma/server.rb:375:in `process_client'
  .gems/gems/puma-2.11.1/lib/puma/server.rb:262:in `block in run'
  .gems/gems/puma-2.11.1/lib/puma/thread_pool.rb:104:in `call'
  .gems/gems/puma-2.11.1/lib/puma/thread_pool.rb:104:in `block in spawn_thread'
```

**Comment by parndt** [comment]
Ah okay, so that's a method we're going to have to abstract.  Any ideas regarding how we can/should build a menu for a Spree user?  Do we require you to add a plugins method or is there some way we can wrap that?

**Comment by bricesanchez** [comment]
I don't know yet but we have to specify that the Spree user is an admin if he can access to the admin resources. Do you have any idea to do that ?

**Comment by bricesanchez** [comment]
May be a Spree admin user should have access to all plugins in Refinery. I think it's how it works in Spree

**Comment by parndt** [comment]
You could probably implement it like the NilUser: https://github.com/refinery/refinerycms/pull/2908/files#diff-25aba864feade964346d74ebc9029e90R15

```ruby
def plugins
  Refinery::Plugins.registered
end
```

**Comment by parndt** [comment]
In fact you could probably just inherit from this NilUser class: https://github.com/refinery/refinerycms/pull/2908/files#diff-25aba864feade964346d74ebc9029e90R1

**Comment by bricesanchez** [comment]
Very interesting, i'm not sure how it works but i can just inherit, i fix all my errors...

**Comment by parndt** [comment]
It basically just says "YES!" to everything when asked whether the user can access something.

**Comment by bricesanchez** [comment]
The "how it works" was for inherit :p

**Comment by parndt** [comment]
Like this:

```ruby
require 'refinery/core/nil_user'

class MyUserAdapter < Refinery::Core::NilUser
end
```

File: core/app/helpers/refinery/authentication_helper.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module Refinery
+  module AuthenticationHelper
+
+    def refinery_user?
+      true
```
Comment (bricesanchez) [inline core/app/helpers/refinery/authentication_helper.rb]: I think with this method, the site_bar is always displayed.

**Comment by parndt** [comment]
Replaced by #2940

**Pull request closed without merging**